WordPress/Classes/Extensions/UIImageView+Gravatar.swift

```diff
@@ -155,4 +209,96 @@ extension UIImageView {
+        if self.isPhotonURL(siteIcon) || self.isWordPressComFilesURL(siteIcon) {
+            return self.siteIconURLForSiteIconUrl(siteIcon)
+        } else if self.isBlavatarURL(siteIcon) {
+            return self.blavatarURLForBlavatarURL(siteIcon)
+        } else {
+            return self.URLForResizedImageURL(siteIcon)
+        }
+    }
+
+    /// Returs the download URL for a square icon with a size of sizeForBlavatarDownload
+    ///
+    /// Parameters:
+    ///  - path: the icon's path
+    fileprivate func siteIconURLForSiteIconUrl(_ path: String) -> URL? {
+        let size = self.sizeForBlavatarDownload()
+        return self.urlForUrlWithFormat(path, format: String(format: "w=%d&h=%d", size, size))
+    }
+
+    /// Sets the blog's icon as content, for private blogs
+    ///
+    /// Parameters:
+    ///  - siteIcon: the icon's path
+    ///  - placeholderImage: the placeholder
+    fileprivate func setImageWithPrivateSiteIcon(siteIcon: String?, placeholderImage: UIImage?) {
+        if siteIcon == nil {
+            return
+        }
+        let imageRequest = PrivateSiteURLProtocol.requestForPrivateSite(from: self.URLWithSiteIcon(siteIcon!))
+        self.setImageWith(imageRequest!, placeholderImage: placeholderImage, success: nil, failure: nil)
+    }
+
+    // MARK: - Private Photon Helpers
+    /// Returs the photon URL for the provided path
+    ///
+    /// Parameters:
+    ///  - urlString: source URL
+    fileprivate func URLForResizedImageURL(_ urlString: String) -> URL? {
+        let size = CGSize(width: UIImageView.BlavatarDefaultSize, height: UIImageView.BlavatarDefaultSize)
+        let url = URL(string: urlString)
+        if url == nil {
+            return nil
+        }
+        return PhotonImageURLHelper.photonURL(with: size, forImageURL: url)
+    }
+
+    // Possible matches are "i0.wp.com", "i1.wp.com" & "i2.wp.com" -> https://developer.wordpress.com/docs/photon/
+    fileprivate func isPhotonURL(_ path: String) -> Bool {
+        return path.range(of: ".wp.com") != nil
+    }
+
+    // MARK: - Blavatar Private Methods
+    /// Returs the icon URL corresponding to the provided path
+    ///
+    /// Parameters:
+    ///  - path: source icon path
+    fileprivate func blavatarURLForBlavatarURL(_ path: String) -> URL? {
+        let size = self.sizeForBlavatarDownload()
+        return self.urlForUrlWithFormat(path, format: String(format: "d=404&s=%d", size))
+    }
+
+    /// Returns the download size for a blavatar
+    fileprivate func sizeForBlavatarDownload() -> Int {
+        var size = UIImageView.BlavatarDefaultSize
+        size *= Int(UIScreen.main.scale)
+
+        return size
+    }
+
+    // MARK: - Other helpers
+    fileprivate func isWordPressComFilesURL(_ path: String) -> Bool {
+        return path.contains(".files.wordpress.com")
+    }
+
+    fileprivate func isBlavatarURL(_ path: String) -> Bool {
+        return path.range(of: "gravatar.com/blavatar") != nil
```

Comment:
You can also use the `.contains("gravatar/blavatar")` snippet as well (so that you get to avoid the nil check!)